src/multithreading.jl

```diff
@@ -0,0 +1,40 @@
+function task_local end
+
+"""
+    task_local(x::T) -> T
+
+Duplicate `x` for a new task such that it can be used concurrently with the original `x`.
```

Comment:
Should we add to the docs that users with custom types might need to dispatch this function on some of their types for threaded assembly?